### ISSUE TITLE
Nextcloud Docker version 15.0.2 notifications not working

### ISSUE THREAD

**Issue by aaronSkar** [nextcloud/talk-android#431]
Hi, I am running Nextcloud version 15.0.2 in docker and am no longer receive notifications on the phone app for conversations/calls.

I also have a Nextcloud 14 snap instance signed in with the same app and calls ring the phone and chat messages create notifications immediately, this version works as expected.

--

NC Version: 15.0.2
NC Talk Version: 5.0.1
NC App Version: 3.4.2
NC Talk App Version: 3.2.1

--
**Expected behavior:**
- Send chat message in 1on1 chat.  Receive notification on receivers phone.
- Start video call in 1on1 chat.  Phone should "ring" until user interacts.

**Actual behavior:**
- Send chat message in 1on1 chat.  Never receive notification on receiver's phone.
- Start video call in 1on1 chat. Notification comes after a very long time (minutes) but similar to message receive notification.  Then when user clicks the notification the "ringing" starts and user can accept or decline call.  
- Note that in either case if you open the app and look at the chat feed things are happening immediately there.  New messages and calls show up very quickly.  It is only the notification that are not working.

**Comment by mario** [comment]
Do you have any errors in the Nc log? Also what version of Android are you running?

**Comment by aaronSkar** [comment]
Hi @mario, I am testing between 2 devices.
Device 1:
Samsung S9 
Android Version 8.0.0
Kernel Version 4.9.65-14745140

Device 2:
Samsung S6
Android Version: 7.0
Kernel Version: 3.10.61-11510561

I have not noticed anything in the logs but ill double check and post back on that.

**Comment by aaronSkar** [comment]
So I do not see any errors. But I did some more testing and have uncovered some interesting behavior.

When testing from an admin account (this account has never been connected to a phone) from the web interface I can message either the S9 or S6 device and notifications and call all function properly.  

But when using either the account on the S6 or S9 notifications do not function properly whether the message is phone to phone or web to phone.  The server was recently upgraded from 14.04 and I had deleted and re-added the accounts on the phone app because of a domain name change. 

I then logged in with the admin account on my phone (S9) and sent a message to the account on the s6 and everything seems to work both ways. So it seems to be specific to accounts that previously were used on the app maybe? Odd thing is that notifications work on the both phones by changing the account on just 1 phone.

**Comment by mario** [comment]
Strange. But I think I know what's happening and I'll look into it.

**Comment by aaronSkar** [comment]
OK thanks!  If you need anything from me just ask and I will try to get the info you need.

**Comment by yohan-b** [comment]
Hi, it seems I have the same issue. This bug makes the Talk Android app mostly useless for me. :-( Is there a workaround while waiting for a fix ? I'd be happy to help with debugging and testing.

**Comment by mario** [comment]
Send me a test account @yohan-b? mario AT nextcloud.com.

**Comment by yohan-b** [comment]
Hi, I've just updated to Nextcloud 15.0.5 and Talk 3.2.7. The bug is still there. :-(

**Comment by aaronSkar** [comment]
I have also performed the same upgrade.  I also did a test migration from a snap instance to a docker instance that that deployment now has similar behavior.

**Comment by aaronSkar** [comment]
On the snap -> docker migration many of the users cannot even log in to android-talk properly. They are only presented with a spinning wheel on white background after granting access.  If the account is removed and re-added it says updating existing account as if the account still exists on the phone.  After uninstalling and re-installing the app seems to have persistent data as it still seems to know of the accounts previously installed.  @mario Do you know of a way to completely delete that data?

**Comment by aaronSkar** [comment]
Ok so one of my iphone users was having the exact same issue.  He noticed that if you logged out of icloud he was able to sign into talk with no issues and notifications work. I will be testing on an android client by signing out of google/samsung/etc services this afternoon and will update if it also solves for android.

**Comment by aaronSkar** [comment]
Ok so I have tried disabling Samsung sync and google app data sync and clearing 3rd party app-data from google. After each step I uninstalled, cleared phone data via android storage and then re-installed the app.  I am able to sign in fine but still no notifications.

**Comment by aaronSkar** [comment]
Ok work around has been found.
I went into mysql manually and deleted all push notification tokens for the user.  Then reinstalled the app and signed in and everything is working normal.  An example for my user on a particular server would be like this:
`delete from oc_notifications_pushtokens where uid like 'aarons';`

**Comment by yohan-b** [comment]
Hi, this won't work for me. The table is already empty :
> select * from oc_notifications_pushtokens;
Empty set (0.00 sec)

**Comment by aaronSkar** [comment]
Hmm.. So I have fixed a handful of users so far by deleting tokens. The app doesn't need to be uninstalled just the user needs to sign back in and then after a few minutes everything is normal. What I find odd is that you don't have any push notifications.  If I delete the token and sign back in with a phone a new token is in the table for that user almost immediately.  I don't think notifications will work without a token in the table that corresponds to the app and user. Maybe @mario can help us out with how notifications work and how the tokens are used but I suspect the lack of tokens is the issue in your case.

**Comment by yohan-b** [comment]
Hi, took me a while, but I finally figured it out. It was not the same bug as aaronSkar. I was using the Talk app from F-Droid, which does not support notifications according to this thread: https://help.nextcloud.com/t/no-call-notification-signal-ring-tone-at-other-party-when-trying-to-start-a-call-audio/34230/6

I downloaded the app from github and now it works. There really should be a warning about this in the app presentation on F-Droid. :-)
